docs/decisions/0002-oauth-api-client-replacement.rst

```diff
@@ -0,0 +1,28 @@
+2. OAuth API Client replacement
+-------------------------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+The ``EdxRestApiClient`` accepts various types of tokens for authentication, which allows for a wide variety of usages. This means that how authentication is performed is leaked into many different applications and is difficult to standardize and change.
+
+Additionally, ``EdxRestApiClient`` made use of the ``slumber`` python library which is no longer supported.
+
+Decision
+--------
+
+We have introduced a new ``OAuthAPIClient`` to replace the now deprecated ``EdxRestApiClient``.  The ``OAuthAPIClient`` can be used for server-to-server calls, accepting a client id and client secret. The underlying implementation of the authentication is meant to be encapsulated and is subject to change.
+
+Because the ``slumber`` python library is no longer supported, it was not used when implementing the ``OAuthAPIClient``.
+
```

Comment:
Is it helpful to mention that `EdxRestApiClient` is a subclass of `requests.Session`, and therefore exposes a simpler interface than slumber?

Reply:
Agreed.  I think you mentioned `OAuthAPIClient` here and above.

I will update the ADR and add some additional docs.